### PR TITLE
[Blog post] Introduce new OTel features in Envoy and Istio

### LINKED ISSUE #4534: New Blog Post: Observability on the Edge: Enhancing OTel support in Envoy and Istio
Creating this issue following the guide here: https://opentelemetry.io/docs/contributing/blog/#before-submitting-a-blog-post

### Short description and outline of your blog post

I contributed several OTel features to both [Envoy](https://github.com/envoyproxy/envoy/pulls?q=is%3Apr+sort%3Aupdated-desc+author%3Ajoaopgrassi+is%3Amerged) and [Istio](https://github.com/istio/istio/pulls?q=is%3Apr+sort%3Aupdated-desc+author%3Ajoaopgrassi+is%3Amerged) and I think it would be cool to share those with the broader community to raise awareness. Those new features are:

- OTLP HTTP Exporter
- Resource detectors (shipped with environment one)
- Sampler API - There's now the ability for anyone to define their samplers

My idea and possible outline is:

- **Where we are now**: Envoy is deprecating OpenTracing and had limited OTel feature support
- **New OTel tracing features Envoy/Istio**: Explain what new features are about and how the community can benefit from it
- Maybe a quick set up demo with screenshots
- **Next up**: Talk about next improvements to be added there, for example align Envoy with the now stable HTTP semconv

### Technologies

- [Envoy](https://www.envoyproxy.io/)
- [Istio](https://istio.io/)
- C++, Go

### Related SIG

I think this spans over multiple, but I'd say the Spec SIG is the most relevant. I talked with @jsuereth as the SIG sponsor which he gladly accepted (thanks!).

### PR DESCRIPTION
**Pull request by joaopgrassi**
- Fixes #4534
- I also added Envoy and Istio to the [integrations list](https://github.com/open-telemetry/opentelemetry.io/tree/main/data/ecosystem/integrations.yaml).
- **Preview**: https://deploy-preview-4553--opentelemetry.netlify.app/blog/2024/new-otel-features-envoy-istio/

### CODE REVIEW

**Review by tiffany76 — COMMENTED** [review]
Just a few small copy edit suggestions. The docs maintainers might have more to say. Thanks!

File: .cspell.yml
Comment (svrnm) [inline .cspell.yml:1]: See @tiffany76's comment: maintain a local ignore list and do not modify the .cspell.yml

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:11]:
```suggestion
microservices effectively is critical. [Kubernetes](https://kubernetes.io/) has become the de facto
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:17]:
```suggestion
[Envoy](https://www.envoyproxy.io/) and [Istio](https://istio.io/), have emerged as the foundation for the smooth management and
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:26]: I prefer introducing the links to projects when they are mentioned for the first time (see my suggestion above), but this is also OK, since you also provide some details on what those projects are. Choose what you prefer.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:32]:
```suggestion
of the [Cloud Native Computing Foundation](https://www.cncf.io/).
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:43]: is there a reason why this is pinned to version 1.29.4?

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:44]:
```suggestion
that do the job of tracing the requests, including the [OpenTelemetry tracer](https://opentelemetry.io/docs/concepts/signals/traces/#tracer).
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:104]:
```suggestion
First, start by installing the [Jaeger operator](https://www.jaegertracing.io/docs/1.57/operator/):
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:182]:
```suggestion
    OTEL_RESOURCE_ATTRIBUTES: "host-name=abc-123"
```

`host-name` is not covered by the semantic conventions, it should be `host.name`, I guess?

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:201]: This assums that people have `k` as an alias for `kubectl`

**Review by svrnm — CHANGES_REQUESTED** [review]

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:5]: Nice, thanks!

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:26]: Yeah I was also in between the two options, but I prefer to link later together with the bit of explanation of what those are.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:43]: I thought about pinning to 1.29 as that is when the new features became available. I guess we could also link to `latest` but that can be misleading if somehow the link changes or whatever else. Not sure what is better 🤔

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:182]: Good catch, I wanted to add an arbitrary attribute but somehow picked host :D. Yes, `host.name` is the semconv correct key. Fixed!

**Comment by joaopgrassi** [comment]
I merged `main`, and `fix:format` is changing format on a lot of files, but here CI only complains about my file. Is there anything that I'm missing locally after merging main into my branch? 🤔

**Comment by svrnm** [comment]
> I merged `main`, and `fix:format` is changing format on a lot of files, but here CI only complains about my file. Is there anything that I'm missing locally after merging main into my branch? 🤔

We updated some of our tooling just yesterday, might be the case that something slipped here, cc @chalin

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:44]: That makes sense, let's keep it pinned then.

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
We just published another blog post, so let's give it another 2 days and then we can merge this one in.

**Comment by chalin** [comment]
All seems to be ok now, right?

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:3]: I like both the long and short titles, but they aren't mutually consistent and I think they should be. I'm not sure how to fix this though.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:54]: Given the context of this post, could we shorten the heading here? E.g.,

```suggestion
## New OTel tracing features in Envoy and Istio
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:63]:
```suggestion
Starting from Envoy 1.29+ and Istio 1.21+, users have access to the new features described below.
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:127]:
```suggestion
Next, install Istio using
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:128]:
```suggestion
[`istioctl`](https://istio.io/latest/docs/setup/install/istioctl/):
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:201]:
```suggestion
To test your setup, make some requests to one of the services, for example:
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:207]:
```suggestion
Then you can check it out on the Jaeger UI -- you should see some traces!
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:222]:
```suggestion

From the spans produced by Envoy you can see (in order):

1. Outgoing (egress) call from the `ratings` service to the
   `productpage` service.
2. Incoming (ingress) call in the `productpage` service.
3. `host-name` resource attribute we applied using the
   `OTEL_RESOURCE_ATTRIBUTES`. This attribute was picked up by the environment
   resource detector and added to all spans Envoy created.

You can also see all the other downstream calls made, as all services
have the Envoy sidecar injected by Istio. You have full observability of the
calls between services, just by enabling the OTel tracer in Envoy!
```

**Review by chalin — COMMENTED** [review]
First pass of copyedits.

**Review by tiffany76 — APPROVED** [review]

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:3]: Yeah looking now I agree with you. I will try to come up with something better.

Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md]: @chalin I tried a small change. Let me know what you think.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:48]:
```suggestion
for tracing requests, including the
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:47]:
```suggestion
Envoy offers several
[HTTP tracers](https://www.envoyproxy.io/docs/envoy/v1.29.4/api-v3/config/trace/trace)
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:49]:
```suggestion
[OpenTelemetry tracer](https://www.envoyproxy.io/docs/envoy/v1.29.4/api-v3/config/trace/v3/opentelemetry.proto). [Tracers](/docs/concepts/signals/traces/#tracer) can be
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:61]:
```suggestion
gRPC, it lacked support for exporting using HTTP. OpenTelemetry supports both
protocols as first-class citizens. In addition, other areas such as providing
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:66]: Because the sentence starts with "Starting" we don't need the "+" signs IMHO.

```suggestion
Starting from Envoy 1.29 and Istio 1.21, users have access to the new features
described below.
```

Comment (chalin) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md:66]: Also, for interested readers who want to find out what is new in those releases, could you provide links to "Envoy 1.29" and "Istio 1.21"?

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:74]: Hmm, maybe:

```suggestion
it to send telemetry to observability sinks using OTLP/HTTP, directly from Envoy
proxies.
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:229]:
```suggestion
With the new features described in this post, users gain more flexibility in exporting their traces.
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:105]:
```suggestion
It's time to see the new features in action! For this, we use the
[Istio Bookinfo application](https://istio.io/latest/docs/examples/bookinfo/),
and illustrate how to:

- Deploy in Kubernetes, with Istio as service mesh
- Export traces to [Jaeger](https://www.jaegertracing.io/) using HTTP
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:156]: Hmm, how about the following?

```suggestion
This installs Istio and configures the OpenTelemetry tracing provider to 
use the `http` exporter over OTLP/HTTP, with the Jaeger
collector as endpoint. This configuration also enables the environment resource detector in
`resource_detectors`.
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:4]:
```suggestion
date: 2024-06-06
```

**Review by chalin — APPROVED** [review]
Very well written post @joaopgrassi! 🙌🏻 I enjoyed reading it.

I made another copyedit pass, see inline comments.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:3]: How about:

```suggestion
title: 'Observability at the Edge: New OTel features in Envoy and Istio'
linkTitle: New OTel features in Envoy and Istio
```

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (chalin) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md]: Yes, better. I've suggested tweaks via https://github.com/open-telemetry/opentelemetry.io/pull/4553/files#r1627377010.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (joaopgrassi) [reply content/en/blog/2024/new-otel-features-envoy-istio/index.md]: Good idea! I added links to the changelog/release notes for each. Also spotted the wrong Istio version, it should be `1.22`. All fixed!

**Comment by chalin** [comment]
Not sure why it's failing. /cc @svrnm 
We may have to do this one manually (I don't have time atm, if someone else wants to give it a try.)

**Comment by joaopgrassi** [comment]
>  We may have to do this one manually (I don't have time atm, if someone else wants to give it a try.)

Seems to have been fixed, I did it manually. :)

**Comment by chalin** [comment]
Ah, it seems to have failed because you pushed a commit before it was able to. I'm glad that it's all fixed.

@svrnm - this seems ready to merge. If it lands today, we'll need to adjust the date.

File: content/en/blog/2024/new-otel-features-envoy-istio/index.md
Comment (svrnm) [inline content/en/blog/2024/new-otel-features-envoy-istio/index.md:4]:
```suggestion
date: 2024-06-07
```

**Comment by svrnm** [comment]
https://opentelemetry.io/blog/2024/new-otel-features-envoy-istio/ 🎉

**Comment by joaopgrassi** [comment]
Thanks for all the help everyone!

**Pull request merged**
